codegen/smithy-dafny-codegen-cli/src/main/java/software/amazon/polymorph/CodegenCli.java

```diff
@@ -89,6 +92,12 @@ private static Options getCliOptions() {
             .longOpt("help")
             .desc("print help message")
             .build())
+          .addOption(Option.builder("r")
+            .longOpt("library-root")
+            .desc("root directory of the library")
+            .hasArg()
+            .required()
+            .build())
```

Comment:
Is this `library-root` the root for the whole project, or the language specific project?

Reply:
The whole project, same as `LIBRARY_ROOT` from the makefiles. That let me get more mileage out of one option because the patch file directory can also just be convention on top of that root.